shared/config_reader/src/config_reader.cc

```diff
@@ -980,7 +980,7 @@ bool ConfigReader::ReadUInt(const char *exp, unsigned int *val) {
   bool ok = lua_isnumber(l_, -1);
   if (ok) {
     *val = static_cast<int>(rint(lua_tonumber(l_, -1)));
-    if (val < 0) {
+    if (rint(lua_tonumber(l_, -1)) < 0) {
```

Comment:
? why call this twice?

Reply:
because if you compare val in new compilers it complains that you're comparing an unsigned int with a signed int. This was a build error that poped up when compiling in ubuntu 22.
I can actually check if I can revert this after the change to the compile flags I did

Reply:
The previous version of the code was actually broken. The check for a negative numeric value needs to come before the number is cast to an *unsigned* int. Casting to unsigned int potentially changes negative values to positive according to twos-complement representation rules, so the test becomes useless.

We should both move the negative value check before the cast and change the cast to `static_cast<unsigned int>`. Right now it is casting to a signed int then assigning that value to an unsigned int lval, which apparently is a valid implicit type conversion but is unnecessarily obfuscated.

As a side note, I think it's ill-advised in this context to accept any floating point value and round to the nearest integer. It would make more sense to complain about non-integer values. (Arguably this minimizes user surprise... if specifying a value of 3.6 doesn't raise an error they would probably not realize it just becomes 4 internally, which could cause a lot of confusion.) But I guess that would be an improvement to check carefully before merging.

Reply:
I moved the check to before the assignment, if there's further desired changes, let's do it in a separate PR

Reply:
(I don't understand. It didn't look like you changed anything since the comments.)